### DISCUSSION TITLE
Resize pin icon

### DISCUSSION THREAD

**Discussion by sylverbee** [Mapsui/Mapsui#2098] [Q&A]
When zooming in, pin icon size decreases, whereas it should increase. I tried to use map resolution but it's not effective on mobile when you pitch the screen. 
Is there any other solution?

**Comment by janusw** [comment]
The pin icon size should not be affected by zooming at all AFAIK.

Maybe you can to give us a bit more detail about what you're actually doing (ideally including a small code sample).

**Reply by sylverbee** [reply]
The general idea is to change pin icon size when zooming in/out.
It's visually nicer and make sence to keep icon the size of the map.

Here is the code I tried (only working on UWP as it gets mouse wheel event)

```
private void MapView_OnZoomed(object sender, ZoomedEventArgs e)
{
    MapControl map = (MapControl)sender;
    foreach (Pin bot in myMap.Pins)
    {
        if (map.Viewport.Resolution >= 5)
        {
            bot.Scale = 0;
        }
        else if (map.Viewport.Resolution > 1 && map.Viewport.Resolution < 5)
        {
            bot.Scale = (float)(1 / map.Viewport.Resolution);
        }
        else
        {
            bot.Scale = (float)(.2 / map.Viewport.Resolution);
        }
    }
}

```

**Reply by janusw** [reply]
> The general idea is to change pin icon size when zooming in/out. It's visually nicer and make sence to keep icon the size of the map.

Well, that does not fit the usual concept of a pin. A pin is commonly assumed to be some simple fixed-size icon that marks a particular point on the map. It does not scale with the map resolution and has no real-world extent on the map. That's how it works on Google Maps, Bing Maps and essentially any other map viewer that I'm aware of.

Of course you can play all kinds of tricks in your custom application and scale the pins as you please. Whether that "makes sense" depends on the use case, I guess.

You could use the `Navigator.ViewportChanged` event to detect changes in zoom level (but note that this will also fire when the map is panned) and use `Pin.Scale` to change the pin size. If you want something that actually covers a fixed (real-world) area on the map, you could use a `Polygon` or `Circle` object.

**Reply by sylverbee** [reply]
Can't get Navigator.ViewportChanged. @janusw have you a code ex. ?

**Comment by charlenni** [comment]
Are you sure, if you use the property Resolution correct? If I'm right, then this is has the unit m/pixel on the equator. If this is the case, then you see only for very zoomed in maps a change.

**Reply by sylverbee** [reply]
didn't understand your proposal
